auth-foundation/src/main/java/com/okta/authfoundation/credential/Credential.kt

```diff
@@ -53,10 +54,9 @@ class Credential internal constructor(
 
     /**
      * The [OidcClient] associated with this [Credential].
-     *
-     * This is the [OidcClient] that should be used for all other operations related to this [Credential].
      */
-    val oidcClient: OidcClient = oidcClient.withCredential(this)
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal val oidcClient: OidcClient = oidcClient.withCredential(this)
```

Comment:
In the Swift `Credential` class, I don't expose methods used to fetch the openIdConfiguration, JWKS information, or other org-wide resources.  

Additionally, a developer can add themselves as a delegate to the client, to receive notifications when requests are made / will be made, when tokens are refreshed, when errors are received, etc.  If the client instance isn't exposed, a developer has no way to add themselves as a delegate.

Reply:
> In the Swift Credential class, I don't expose methods used to fetch the openIdConfiguration, JWKS information, or other org-wide resources.

I envision usages such as these to use `CredentialBootstrap.oidcClient` rather than `credential.oidcClient`.

> Additionally, a developer can add themselves as a delegate to the client, to receive notifications when requests are made / will be made, when tokens are refreshed, when errors are received, etc. If the client instance isn't exposed, a developer has no way to add themselves as a delegate.

These aren't mutable properties in the kotlin SDK. You need to specify these things in the configuration before creating the `OidcClient`.